### PR TITLE
Remove NOJSC macro.

### PR DESCRIPTION
**Pull request by JunielKatarn**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

#### Description of changes

Remove macro `NOJSC`, no longer used by react-native-windows.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native/pull/94)

### CODE REVIEW

File: ReactCommon/cxxreact/Platform.h [hunk 1]
```diff
@@ -18,7 +18,7 @@
 namespace facebook {
 namespace react {
 
-#if (!defined (NOJSC)) && (!V8_ENABLED)
+#if (!_MSC_VER) && (!V8_ENABLED)
 namespace JSCNativeHooks {
```
Comment (rohitjain85) [inline ReactCommon/cxxreact/Platform.h]: Does anyone need this? This is not needed on droid and windows and I believe Apple is already moved to JSI flow? 

Namrata who is interning in my team had raised this PR where she is getting rid of this block.

[https://github.com/microsoft/react-native/pull/93](https://github.com/microsoft/react-native/pull/93)

Comment (mganandraj) [reply ReactCommon/cxxreact/Platform.h]: Do we have any build configuration (OSS/SDX_PLAT/DEVMAIN) which still builds droid with JSC ?

---
In reply to: [297099783](https://github.com/microsoft/react-native/pull/94#discussion_r297099783) [](ancestors = 297099783)

File: ReactCommon/make.inc [hunk 2]
```diff
@@ -55,7 +55,4 @@ C_DEFINES = $(C_DEFINES) -D NOMINMAX
 C_DEFINES = $(C_DEFINES) -D WITH_OFFICE_TRACING=0
 !endif
 
-NOJSC = 1;
```
Comment (mganandraj) [inline ReactCommon/make.inc]: Hope you have verfied devmain build too ..

**Comment by JunielKatarn** [comment]
Superseded by 53753d082ebcc3b2d74d3e5d39cbd8076097553d.

Closing this PR.

**Pull request closed without merging**